### PR TITLE
Add basic methods for tagging in ActsAsTaggableOn

### PR DESCRIPTION
**Pull request by lpichler**
this PR is adding those methods:

**instance methods:**
- tag_list
- tag_add
- is_tagged_with?
- tag_remove

**class methods:**
- tagged_with 

## Links
https://projects.engineering.redhat.com/browse/TPINVTRY-184

### CODE REVIEW

File: app/models/concerns/act_as_taggable_on.rb [hunk 1]
```diff
@@ -21,6 +21,64 @@ def taggings
           }
         end
       end
+
+      def tag_list
+        klass = self.class
+        klass.joins(klass.tagging_relation_name => :tag).references(klass.tagging_relation_name => :tags).where(:id => id).group(:id)&.take&.tags&.pluck(:name) || []
```
Comment (agrare) [inline app/models/concerns/act_as_taggable_on.rb]: Can't this just be simply `tags.pluck(:name)` ?

Comment (agrare) [reply app/models/concerns/act_as_taggable_on.rb]: :+1: nice

**Comment by agrare** [comment]
@lpichler really like how this is looking, is it still wip or is this good to go?

**Comment by lpichler** [comment]
@agrare I pushed now some other changes(hopefully positive :) ), not it is ready for review 👍

**Comment by lpichler** [comment]
thanks @agrare !

**Pull request merged**